File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/network/ui/NetWorkMainPagerAdapter.java
package com.didichuxing.doraemonkit.kit.network.ui;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;
import androidx.viewpager.widget.PagerAdapter;

import com.didichuxing.doraemonkit.R;

import java.util.ArrayList;
import java.util.List;
public class NetWorkMainPagerAdapter extends PagerAdapter {
    private List<View> views;
    private List<String> tabs = new ArrayList<>();

    public NetWorkMainPagerAdapter(Context context, List<View> views) {
        this.views = views;
        tabs.add(context.getString(R.string.dk_net_monitor_title_summary));
        tabs.add(context.getString(R.string.dk_net_monitor_list));
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return tabs.get(position);
    }

    @Override
    public int getCount() {
        return views.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        View view = views.get(position);
        container.addView(view);
        return view;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView(views.get(position));
    }

}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/network/ui/NetWorkMainPagerAdapter.kt
package com.didichuxing.doraemonkit.kit.network.ui

import android.content.Context
import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter
import com.didichuxing.doraemonkit.R
import java.util.*

class NetWorkMainPagerAdapter(context: Context?, private val views: List<View>) : PagerAdapter() {
    private val tabs: MutableList<String> = ArrayList()
    override fun getPageTitle(position: Int): CharSequence? {
        return tabs[position]
    }

    override fun getCount(): Int {
        return views.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = views[position]
        container.addView(view)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(views[position])
    }

    init {
        tabs.add(context!!.getString(R.string.dk_net_monitor_title_summary))
        tabs.add(context.getString(R.string.dk_net_monitor_list))
    }
}